### DISCUSSION TITLE
the install.sh script does not create hbbs and hbbr

### DISCUSSION THREAD

**Discussion by gattolfo98** [rustdesk/rustdesk#4539] [Q&A]
I'm trying to install on my old rustdesk home server, but I do the intallation script tells me:
`'/opt/rustdesk/hbbs': File or directory does not exist`
`'/opt/rustdesk/hbbr': File or directory does not exist`

it's normal? in addition, if I say to use IP instead of domain, it sets me an IP outside my subnet

**Comment by dinger1986** [comment]
this is presumably the script by techahold? what are you installing on?

**Reply by CastOak** [reply]
I am having the same issue, installing on a dietpi through ssh. wget of the techahold install.sh goes flawless as does the update of package lists. script then runs the chmod commands and hangs when it doesn't find /opt/rustdesk/hbbs . 
Could have sworn I used this script in the past to get this running on the same platform, but now I am just stucker than a stuck thing.

**Reply by dinger1986** [reply]
maybe missing a dependency? curl, tar, wget, maybe even sudo could be one of them?

**Reply by CastOak** [reply]
yeah that was my first thought too, however it does not appear to be according to the script output:

`xxxx@DietPi:~# sudo ./install.sh
Installing prerequisites
Get:1 http://raspbian.raspberrypi.org/raspbian bullseye InRelease [15.0 kB]
Get:2 https://archive.raspberrypi.org/debian bullseye InRelease [23.6 kB]
Get:3 https://archive.raspberrypi.org/debian bullseye/main armhf Packages [313 kB]
Fetched 352 kB in 9s (41.2 kB/s)
Reading package lists... Done
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
curl is already the newest version (7.74.0-1.3+deb11u11).
dnsutils is already the newest version (1:9.16.44-1~deb11u1).
tar is already the newest version (1.34+dfsg-1).
unzip is already the newest version (6.0-26+deb11u1).
wget is already the newest version (1.21-1+deb11u1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1) IP
2) DNS/Domain
Choose your preferred option, IP or DNS/Domain:1
Installing Rustdesk Server
chmod: cannot access '/opt/rustdesk/hbbs': No such file or directory
chmod: cannot access '/opt/rustdesk/hbbr': No such file or directory`

**Reply by dinger1986** [reply]
Ok, run it line by line, there's got to be something missing, maybe the /opt folder doesn't exist?

I don't have dietpi so guessing but gotta be a way to do it

**Reply by CastOak** [reply]
alright thanks for the advice, will do and get back. 
(DietPi is basically Debian)

**Reply by dinger1986** [reply]
Thanks, yeah weird though because it runs on Debian fine, possibly the sudo command isn't available?

No wait you ran it as sudo so not sudo.

**Reply by CastOak** [reply]
yeah I think I got it. Maybe you could help me understand the script a little bit? It has to do with the ${ARCH} variable - I can't see where that is set other than = uname at the beginning. 

Then in lines 132-143 in install.sh it goes through an elif for ${ARCH} and downloads any one of three install packages. 

HOWEVER, if the elif doesn't find a match for ${ARCH} (i.e. it doesn't know the correct package to download for the OS) it just exits without warning that it hasn't installed anything and continues to `chmod +x /opt/rustdesk/hbbs` which for obvious reasons fails if nothing was installed.

It fails on DietPi because it is a raspberry pi build and it wants the install package [https://github.com/rustdesk/rustdesk/releases/download/1.1.9/rustdesk-1.1.9-raspberry-armhf.deb], so when the script arrives at line 146 the /opt/rustdesk/ folder is empty.

**Reply by dinger1986** [reply]
So what is your output of uname?

**Reply by dinger1986** [reply]
If it'll work with the deb file why not just install that?

Wget to get the the file and then dpkg -i filename. Might need to do apt install -f

**Reply by CastOak** [reply]
uname = Linux

well, it would be "nice" if the install script worked as it is a bit less fiddly (and makes it more approachable for "normal" users), also I just got sucked into figuring it out :) 

think I will just go ahead and do the three .deb pkgs for now :) thanks for your input

**Reply by CastOak** [reply]
just in case anyone else stumbles across this thread, the deb packages in question relevant to DietPi are:

For the RustDesk server:
https://github.com/rustdesk/rustdesk-server/releases/download/[current_version]/rustdesk-server-utils_[current_version]_armhf.deb
https://github.com/rustdesk/rustdesk-server/releases/download/[current_version]/rustdesk-server-hbbs_[current_version]_armhf.deb
https://github.com/rustdesk/rustdesk-server/releases/download/[current_version]/rustdesk-server-hbbr_[current_version]_armhf.deb

For the RustDesk client:
https://github.com/rustdesk/rustdesk/releases/download/[current_version]/rustdesk-[current_version]-raspberry-armhf.deb

**Reply by dinger1986** [reply]
Uname Linux unfortunately is too generic.

If it had identified as Debian it would have worked fine.
